includes/Framework/Utilities/BackgroundJobHandler.php

```diff
@@ -127,20 +127,35 @@ public function maybe_handle() {
 			wp_die();
 		}
 
-		// WC core does 2 things here that can interfere with our nonce check:
-		// 1. WooCommerce starts a session due to our GET request to dispatch a job
-		//  However, this happens *after* we've generated a nonce without a session (in CRON context)
-		// 2. it then filters nonces for logged-out users indiscriminately without checking the nonce action; if
-		//  there is a session created (and now the server does have one), it tries to filter every.single.nonce
-		//  for logged-out users to use the customer session ID instead of 0 for user ID. We *want* to check
-		//  against a UID of 0 (since that's how the nonce was created), so we temporarily pause the
-		//  logged-out nonce hijacking before standing aside.
-		remove_filter( 'nonce_user_logged_out', array( WC()->session, 'nonce_user_logged_out' ) );
+		/**
+		 * WC core does 2 things here that can interfere with our nonce check:
+		 *
+		 * 1. WooCommerce starts a session due to our GET request to dispatch a job
+		 *  However, this happens *after* we've generated a nonce without a session (in CRON context)
+		 * 2. it then filters nonces for logged-out users indiscriminately without checking the nonce action; if
+		 *  there is a session created (and now the server does have one), it tries to filter every.single.nonce
+		 *  for logged-out users to use the customer session ID instead of 0 for user ID. We *want* to check
+		 *  against a UID of 0 (since that's how the nonce was created), so we temporarily pause the
+		 *  logged-out nonce hijacking before standing aside.
+		 *
+		 * @see \WC_Session_Handler::init() when the action is hooked
+		 * @see \WC_Session_Handler::nonce_user_logged_out() WC < 5.3 callback
+		 * @see \WC_Session_Handler::maybe_update_nonce_user_logged_out() WC >= 5.3 callback
+		 */
+		if ( Compatibility::is_wc_version_gte('5.3') ) {
+			$callback = [ WC()->session, 'maybe_update_nonce_user_logged_out' ];
```

Comment:
One small thing. Can we keep the array syntax consistent with the phpcs ruleset?

Reply:
Nice. Thanks @ibndawood